python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -0,0 +1,30 @@
+from typing import Sequence, Union
+
+import dagster._check as check
+from dagster._core.selector.subset_selector import (
+    generate_asset_dep_graph,
+    generate_asset_name_to_definition_map,
+)
+
+from .assets import AssetsDefinition
+from .source_asset import SourceAsset
+
+
+class AssetGraph:
```

Comment:
would be nice to have some convenient properties like `upstream_mapping` and `downstream_mapping` (from the underlying asset_dep_graph, but that can come later